### PR TITLE
Bugfix/cs840 fix calc rebased

### PR DESCRIPTION
**Pull request by kuenishi**
addresses #840 (c.f. #862)

### CODE REVIEW

**Comment by shino** [comment]
Still xref fails

```
src/riak_cs_utils.erl:518: Warning maybe_process_resolved/3 calls undefined function riak_object:bucket/1
src/riak_cs_utils.erl:518: Warning maybe_process_resolved/3 calls undefined function riak_object:get_contents/1
src/riak_cs_utils.erl:518: Warning maybe_process_resolved/3 calls undefined function riak_object:key/1
```

**Comment by shino** [comment]
- [ ] Eunit and eqc tests pass
- [x] Dialyzer output clean (or no new warnings)
- [x] xref
- [x] `riak_test` tests pass

File: riak_test/tests/storage_stats_test.erl [hunk 1]
```diff
@@ -48,6 +69,59 @@ confirm() ->
...
+
+    Pid = rt:pbc(hd(RiakNodes)),
+
+    store_object(?BUCKET4, UserConfig),
+    store_object(?BUCKET5, UserConfig),
+    store_object(?BUCKET6, UserConfig),
+    %% bucket7 should be counted
+    store_object(?BUCKET8, UserConfig), %% bucket 8 should no be counted
+
+    %% None of thes objects should not be calculated effective in storage
```
Comment (shino) [inline riak_test/tests/storage_stats_test.erl:81]: - s/thes/these/
- not none, the last one (active/undefined) should be calculated (as your comment on line 84)

File: riak_test/tests/storage_stats_test.erl [hunk 2]
```diff
@@ -48,6 +69,59 @@ confirm() ->
...
+    %% tombstone
 (see above adding {delete_mode, keep} to riak_kv
+    Bucket = <<"0o:", (crypto:md5(list_to_binary(?BUCKET8)))/binary>>,
+    {ok, RiakObject} = riakc_pb_socket:get(Pid, Bucket, list_to_binary(?KEY)),
+    %% This leaves a tombstone
+    ok = riakc_pb_socket:delete_obj(Pid, RiakObject),
+
+    ok = riakc_pb_socket:stop(Pid),
+    ok.
+    %% garbage end
```
Comment (shino) [inline riak_test/tests/storage_stats_test.erl]: Need this comment?

File: riak_test/tests/storage_stats_test.erl [hunk 3]
```diff
@@ -28,26 +28,132 @@
...
+%% makes manifest in BUCKET(4,5,6,7,8) to garbage, which can
+%% be generated from former versions of riak cs than 1.4.5
+verify_cs840_regression(UserConfig, RiakNodes) ->
+
+    Pid = rt:pbc(hd(RiakNodes)),
+
+    %% bucket7 should be counted
+
+
+    %% %% None of thes objects should not be calculated effective in storage
```
Comment (shino) [inline riak_test/tests/storage_stats_test.erl]: The object in BUCKET7 should be counted.

File: riak_test/tests/storage_stats_test.erl [hunk 4]
```diff
@@ -28,26 +28,132 @@
...
+                                   %% state=active, .props=undefined
+                                   {?BUCKET7, ?KEY, active, undefined}]),
+
+    ok = mess_with_tombstone(Pid, UserConfig),
+    ok = riakc_pb_socket:stop(Pid),
+    ok.
+
+mess_with_tombstone(Pid, UserConfig) ->
+    %% %% tombstone
 (see above adding {delete_mode, keep} to riak_kv
```
Comment (shino) [inline riak_test/tests/storage_stats_test.erl]: Tow `%%`s

Comment (shino) [inline riak_test/tests/storage_stats_test.erl]: Open parenthesis has no partner :sweat:

File: riak_test/tests/storage_stats_test.erl [hunk 5]
```diff
@@ -28,26 +28,132 @@
...
+                Manifest1 = Manifest0?MANIFEST{state=NewState, props=Props, content_length=10},
+                RiakObject = riakc_obj:update_value(RiakObject0,
+                                                    term_to_binary([{UUID, Manifest1}])),
+                lager:info("~p", [Manifest1?MANIFEST.props]),
+
+                Block = crypto:rand_bytes(100),
+                ?assertEqual([{version_id, "null"}], erlcloud_s3:put_object(CSBucket, CSKey,
+                                                                            Block, UserConfig)),
+                ok = riakc_pb_socket:put(Pid, RiakObject),
+                assure_num_siblings(Pid, Bucket, list_to_binary(CSKey), 2)
```
Comment (shino) [inline riak_test/tests/storage_stats_test.erl]: Is it better to have siblings here?
For, at least, props=undefined case, just single (no siblings) manifest with props=undefined seems straightforward for me, doesn't it?

Comment (kuenishi) [inline riak_test/tests/storage_stats_test.erl]: I cut out active/undefined case to another function and removed siblings at 2ce2686.

**Comment by shino** [comment]
:+1: for mege

**Comment by kuenishi** [comment]
Thanks!

**Pull request merged**